src/output/file.rs

```diff
@@ -0,0 +1,289 @@
+        }
+    }
+
+    /// Builds a new File object using [`FileBuilder`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ocptv::output::*;
+    ///
+    /// let file = File::builder("name", "/path/to/file", false)
+    ///     .description("description")
+    ///     .content_type("text/plain")
+    ///     .add_metadata("key", "value".into())
+    ///     .build();
+    /// ```
+    pub fn builder(name: &str, uri: &str, is_snapshot: bool) -> FileBuilder {
+        FileBuilder::new(name, uri, is_snapshot)
+    }
+
+    /// Creates an artifact from a File object.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ocptv::output::*;
+    ///
+    /// let file = File::new("name", "/path/to/file", false);
+    /// let _ = file.to_artifact();
+    /// ```
+    pub fn to_artifact(&self) -> spec::File {
+        spec::File {
+            name: self.name.clone(),
+            uri: self.uri.clone(),
+            is_snapshot: self.is_snapshot,
+            description: self.description.clone(),
+            content_type: self.content_type.clone(),
+            metadata: self.metadata.clone(),
+        }
+    }
+}
+
+/// This structure builds a [`File`] object.
+///
+/// # Examples
+///
+/// ```
+/// # use ocptv::output::*;
+///
+/// let builder = File::builder("name", "/path/to/file", false)
+///     .description("description")
+///     .content_type("text/plain")
+///     .add_metadata("key", "value".into());
+/// let file = builder.build();
+/// ```
+#[derive(Default)]
+pub struct FileBuilder {
+    name: String,
+    uri: String,
+    is_snapshot: bool,
+    description: Option<String>,
+    content_type: Option<String>,
+    metadata: Option<BTreeMap<String, tv::Value>>,
+}
+
+impl FileBuilder {
+    /// Creates a new FileBuilder.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ocptv::output::*;
+    ///
+    /// let builder = FileBuilder::new("name", "/path/to/file", false);
+    /// ```
+    pub fn new(name: &str, uri: &str, is_snapshot: bool) -> Self {
+        FileBuilder {
+            name: name.to_string(),
+            uri: uri.to_string(),
+            is_snapshot,
+            ..Default::default()
+        }
+    }
+
+    /// Add a description to a [`FileBuilder`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ocptv::output::*;
+    ///
+    /// let builder = FileBuilder::new("name", "/path/to/file", false)
+    ///     .description("description");
+    /// ```
+    pub fn description(mut self, description: &str) -> FileBuilder {
+        self.description = Some(description.to_owned());
+        self
+    }
+
+    /// Add a content_type to a [`FileBuilder`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ocptv::output::*;
+    ///
+    /// let builder = FileBuilder::new("name", "/path/to/file", false)
+    ///     .content_type("text/plain");
+    /// ```
+    pub fn content_type(mut self, content_type: &str) -> FileBuilder {
```

Comment:
spec says this is a MIME type, not an arbitrary string; 

example of what could be used: https://crates.io/crates/mime
```suggestion
    pub fn content_type(mut self, content_type: mime::Mime) -> FileBuilder {
```